crates/rover-client/src/operations/subgraph/routing_url/runner.rs

```diff
@@ -0,0 +1,116 @@
+use super::types::*;
+use crate::blocking::StudioClient;
+use crate::operations::config::is_federated::{self, IsFederatedInput};
+use crate::RoverClientError;
+
+use graphql_client::*;
+
+#[derive(GraphQLQuery)]
+// The paths are relative to the directory where your `Cargo.toml` is located.
+// Both json and the GraphQL schema language are supported as sources for the schema
+#[graphql(
+    query_path = "src/operations/subgraph/routing_url/routing_url_query.graphql",
+    schema_path = ".schema/schema.graphql",
+    response_derives = "Eq, PartialEq, Debug, Serialize, Deserialize",
+    deprecated = "warn"
+)]
+/// This struct is used to generate the module containing `Variables` and
+/// `ResponseData` structs.
+/// Snake case of this name is the mod name. i.e. subgraph_routing_url_query
+pub(crate) struct SubgraphRoutingUrlQuery;
+
+/// Fetches a subgraph's routing URL (String)
+pub fn run(
+    input: SubgraphRoutingUrlInput,
+    client: &StudioClient,
+) -> Result<Option<String>, RoverClientError> {
```

Comment:
```suggestion
) -> Result<String, RoverClientError> {
```

I think this should be a `String` and not an `Option<String>` because a subgraph _must_ have a routing URL if it exists in GraphOS. Therefore, if one _doesn't_ exist, that is an error (probably subgraph doesn't exist or malformed response)

Reply:
👍 35d6dc4